### ISSUE TITLE
Provide more details for the errors

### ISSUE THREAD

**Issue by pcibraro** [Azure/azure-storage-ruby#27]
In many cases, you get the generic error "one of the request inputs is not valid", which is returned by the REST API, but the API also returns a detail of which field or input is not valid. I am not seeing that being returned in the exception message.

**Comment by yaxia** [comment]
@pcibraro Could you please elaborate about what APIs you are calling? Most of the HTTP errors will be thrown with detail information:

``` ruby
require "azure/storage"
client = Azure::Storage.client(
  :storage_account_name => <account_name>, 
  :storage_access_key => <key>)

container_name = 'anonexistingcontainername'
begin
  client.blobClient.get_container_properties container_name
rescue Azure::Core::Http::HTTPError => ex
  puts "Status: #{ex.status_code}, Description: #{ex.description}"
  puts "Exception: #{ex}"
end
```
Outputs:
```
Status: 404, Description: The specified container does not exist.
RequestId:61c86b80-0001-000d-48dc-d1790b000000
Time:2016-06-29T07:59:31.4008044Z
Exception: ContainerNotFound (404): The specified container does not exist.
RequestId:61c86b80-0001-000d-48dc-d1790b000000
Time:2016-06-29T07:59:31.4008044Z
```

**Comment by pcibraro** [comment]
Yes, try with this one,
```
blob_service.put_blob_pages(container_name, blob_name, 0, 512, page_data, {})
```
This gives a page of 513 bytes, which is invalid. Azure returns that the Range is invalid, but it's not reflected in the exception.

**Comment by vinjiang** [comment]
@yaxia , let's follow-up.

**Comment by yaxia** [comment]
Fixed with the release of azure-core@0.1.4.
